internal/batch/batch_manager.go

```diff
@@ -343,10 +347,14 @@ func (bm *batchManager) getProcessors() []*batchProcessor {
 	bm.dispatcherMux.Lock()
 	defer bm.dispatcherMux.Unlock()
 
+	exists := make(map[*batchProcessor]bool)
 	var processors []*batchProcessor
 	for _, d := range bm.dispatchers {
 		for _, p := range d.processors {
-			processors = append(processors, p)
+			if !exists[p] {
+				processors = append(processors, p)
```

Comment:
Seems fine... but why does it contain duplicates in the first place? Guess we can have a processor shared by more than one dispatcher?

Reply:
```go
for _, d := range bm.dispatchers {
```
This is iterating over a map, where a dispatcher can register itself multiple times.

I've updated the code to maintain a list as well as a map, to avoid this confusion

Reply:
Makes sense, thanks for the tweak